scripts/bert/embedding.py

```diff
@@ -36,44 +42,59 @@ class BertEmbedding(object):
     ----------
     ctx : Context.
         running BertEmbedding on which gpu device id.
+    dtype: str
+        data type to use for the model.
     model : str, default bert_12_768_12.
         pre-trained BERT model
     dataset_name : str, default book_corpus_wiki_en_uncased.
         pre-trained model dataset
+    params_path: str, default None
+        path to a parameters file to load instead of the pretrained model.
     max_seq_length : int, default 25
         max length of each sequence
     batch_size : int, default 256
         batch size
     """
 
-    def __init__(self, ctx=mx.cpu(), model='bert_12_768_12',
+    def __init__(self, ctx=mx.cpu(), dtype='float32', model='bert_12_768_12',
                  dataset_name='book_corpus_wiki_en_uncased',
-                 max_seq_length=25, batch_size=256):
+                 params_path=None, max_seq_length=25, batch_size=256):
         """
         Encoding from BERT model.
 
         Parameters
         ----------
         ctx : Context.
             running BertEmbedding on which gpu device id.
+        dtype: str
+            data type to use for the model.
         model : str, default bert_12_768_12.
             pre-trained BERT model
         dataset_name : str, default book_corpus_wiki_en_uncased.
             pre-trained model dataset
+        params_path: str, default None
+            path to a parameters file to load instead of the pretrained model.
         max_seq_length : int, default 25
             max length of each sequence
         batch_size : int, default 256
             batch size
         """
         self.ctx = ctx
+        self.dtype = dtype
         self.max_seq_length = max_seq_length
         self.batch_size = batch_size
+        self.dataset_name = dataset_name
         self.bert, self.vocab = gluonnlp.model.get_model(model,
-                                                         dataset_name=dataset_name,
+                                                         dataset_name=self.dataset_name,
                                                          pretrained=True, ctx=self.ctx,
                                                          use_pooler=False,
```

Comment:
Should we also let user decide these value?

Reply:
Are you referring to the `use_pooler` parameter? I don't think it makes sense since this class is only concerned with getting embeddings for a certain input sequence.

Reply:
You are right. I was thinking to get the bert model in user's need. But it is not necessary here.

Reply:
Eventually it might be nice to move the `BertEmbedding` class and other similar ones that are in these scripts to the main gluonnlp package. Makes it more obvious for an end-user that they exist, and they can be tested more thoroughly there. I'd leave that for a later PR though :)